### DISCUSSION TITLE
Dynamic Module Federation on Angular Universal (SSR)

### DISCUSSION THREAD

**Discussion by rpanadero** [nrwl/nx#23432] [Feature Requests]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
I would like to request that Dynamic Module Federation works on Angular Universal. I know there are no plans to support it after researching into issues: https://github.com/nrwl/nx/issues/10887#issuecomment-1428093583. However, I think that this should be reconsidered

Static module federation is not a workaround for me because I need to have different remote URLs per environment (dev, pre and pro stages) without building every time. 

As far as I know, the problem happens when "loadRemoteModule" function is called and internally tries to dynamically import the remote module by doing something like this:

```
import("http://my-remote-module/remoteEntry.js")
```

## Motivation
Nowadays there are many applications that require server side rendering for SEO optimization and not to support dynamic module federation in this scenario is a little bit ugly.

## Suggested Implementation
N/A

## Alternate Implementations
N/A

**Comment by luizluan** [comment]
I also need this, most of my application are external plugins.

**Comment by ArgV04** [comment]
Maybe something you can take from here?

https://github.com/module-federation/module-federation-examples/tree/master/angular-universal-ssr

**Comment by johnitvn** [comment]
i am also trying to do this the result i get when run serve —devRemotes always 404 remoteEntry.mjs

**Comment by Coly010** [comment]
Hey :) Trust me when I say, I hear you 🙂 
This is something that we want to add and to get right. SSR is fast becoming a first-class citizen in the Frontend World in general and we want to ensure that our Module Federation support for it is as great as possible. 

It's something that we're tracking internally and it will be addressed at some point in the future. There are a lot of nuances with it (node vs browser) and therefore it is a little bit tricky.

**Comment by sangaiah** [comment]
Hi I am also waiting for this,
Currently I am using "@nx/angular": "19.0.1" with dynamic modulefedration. Looking for Server Side Rendering. 

As of now, getting below error while executing "function loadModule(url){return import(url)}"
 _ERROR TypeError [ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING]: A dynamic import callback was not specified.
    at importModuleDynamicallyCallback (node:internal/modules/esm/utils:228:9)_

Any workaround to resolve this issue?

**Reply by rpanadero** [reply]
Add logic to your application to avoid microfrontend rendering on server side
